spark/v3.4/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestStoragePartitionedJoinsInRowLevelOperations.java

```diff
@@ -257,14 +261,22 @@ private void checkMerge(RowLevelOperationMode mode) {
           SparkPlan plan =
               executeAndKeepPlan(
                   "MERGE INTO %s AS t USING %s AS s "
-                      + "ON t.id = s.id AND t.dep = s.dep AND t.dep = 'hr'"
```

Comment:
This condition was a hard requirement in 3.3, otherwise SPJ would not kick in. It is no longer needed, so I adapted the test to verify mismatching keys are supported. That's the usual MERGE use case since it may contain data for new partitions.

Reply:
Should we leave in a test that does this join though? I know it shouldn't be required but we might as well keep a test that also checks if SPJ kicks in with the additional predicate as well.

Reply:
I don't think a predicate would affect SPJ but I can keep this test separately to be safe.

Reply:
Added both tests now.